### ISSUE TITLE
cant' open a .py file with Pyzo and Pyzo crashed if config.ssdf is removed 

### ISSUE THREAD

**Issue by operade** [pyzo/pyzo#946]
Hi I have just installed Pyzo with Homebrew on a Mac on which I did a clean install of macOS Sonoma 14.2.1 and two things are stranged

first : when I click on file say example.py a pyzo window appears with nothing except an empty shell. In can use the "open file" option in the menu to do it, but I didn't have to do that before.

second : I tried to remove the config file "config.ssdf" which is in the hidden directory .pyzo : bad idea because pyzo crashed although I was expected that the config file will be rebuilt. If I uninstall with homebrew and then reinstall Pyzo (without putting the config file) it is the same. I join a report file.
[report.txt](https://github.com/pyzo/pyzo/files/13805451/report.txt)

What did I did wrong ? Any help would be appreciated. Thanks in advance and Happy New Year.

David

**Comment by almarklein** [comment]
When I try to open a `.py` file on my Mac by double-clicking, Pyzo opens but does not open the file. So it looks like something is broken there 🤔  - I did change something related recently: #923

When the "config.ssdf" is deleted, Pyzo should indeed revert to its defaults and overwrite the config file when it exists. The crash report suggests that it segfaults, which is pretty bad and is usually related to Qt.

Have you tried the binary from Github, or running Pyzo from source?

**Comment by operade** [comment]
Hi Installing Pyzo with the .dmg file is the same for both issues; I haven't tried the Github binary. However, at the beginning I wanted to change the config.ssdf file because I was unable to change the path where Python is located on my Mac in the Pyzo preferences pane : if I did so, when Pyzo restarted the change was not made. I was able to resolve the issue by changing the owner of the file from "root" to "myself". Best regards

**Comment by almarklein** [comment]
So the crash is fixed, and the remaining issue is that double-clicking a file in the OS file explorer does not open the file in Pyzo, am I right?

**Comment by operade** [comment]
Yes absolutely : sorry for my late reply

**Comment by jowe19** [comment]
Hi. 
I've got the same problem ; if I try to open a `py` file on my Mac by double-clicking, it launches Pyzo but without opening the file.

**Comment by bdieterm** [comment]
@jowe19
I don't have a Mac, but you could try the following to hopefully get more information about the problem:
Start Pyzo normally.
Open the "Logger" tool (via "Tools -> Logger" in Pyzo's menu).
Double click a Python file to open it.
Normally the Logger will show some text like:
Request: /path/to/file.py
Reply: Try opening file ...

What does the Logger show in your case?

**Comment by jowe19** [comment]
> @jowe19 I don't have a Mac, but you could try the following to hopefully get more information about the problem: Start Pyzo normally. Open the "Logger" tool (via "Tools -> Logger" in Pyzo's menu). Double click a Python file to open it. Normally the Logger will show some text like: Request: /path/to/file.py Reply: Try opening file ...
> 
> What does the Logger show in your case?

Thanks for your help, but after a double-lick the Logger doesn't change : 
<img width="999" alt="Capture d’écran 2024-02-02 à 16 04 36" src="https://github.com/pyzo/pyzo/assets/19306526/e7962ff3-ae71-4080-b0e2-ffe8cf24666b">

**Comment by bdieterm** [comment]
@jowe19 
Ok, so there was no request.
You could try to add three lines to Pyzo's source code and then do the same sequence again:

in pyzo/_start.py, after line
`    # Handle command line args now`
insert
`    print('handle_cmd_args in _start.py', sys.argv)`

in pyzo/_start.py, after line
`            fname = str(event.file())`
insert
`            print('handle_cmd_args in QFileOpenEvent', fname, sys.argv)`

in pyzo/core/main.py, after line
`        # Handle any actions`
insert
`        print('handle_cmd_args in main.py', sys.argv)`

**Comment by operade** [comment]
Thank you bdieterm for your help ; 

Where are pyzo/_start.py and  pyzo/core/main.py located on a Mac ? I specify that I display hidden files

**Comment by operade** [comment]
Hi bdieterm thank you for your help after adding the three lines I got an error 
[Capture d’écran.pdf](https://github.com/pyzo/pyzo/files/14302729/Capture.d.ecran.pdf)

**Comment by bdieterm** [comment]
That error means that there are "tabs" used in line 72 in file "pyzo/_start.py".
Depending on text editor, pressing the tab key (at the beginning of a line) or return key (at the end of a line) might insert wider spaces contrary to normal spaces when pressing the space bar.
To fix that, open the file with a text editor, remove all the spaces and tabs at the beginning of line 72, then re-type the spaces using only the space bar on the keyboard, and finally save the file again.

Here is an extract of _start.py, just to make sure you have the correct number of spaces after editing:
```python
class MyApp(QtWidgets.QApplication):
    """So we an open .py files on OSX.
    OSX is smart enough to call this on the existing process.
    """

    def event(self, event):
        if isinstance(event, QtGui.QFileOpenEvent):
            fname = str(event.file())
            print('handle_cmd_args in QFileOpenEvent', fname, sys.argv)
            if fname and fname != "pyzo":
                sys.argv[-1:] = [fname]
                res = commandline.handle_cmd_args()
                if not commandline.is_our_server_running():
                    print(res)
                    sys.exit()
        return QtWidgets.QApplication.event(self, event)
```

**Comment by operade** [comment]
Ok thanks : I think I did the changes correctly ; here is what is showed in the logger  when I open the file "CCPModPC15.py" : Pyzo is launched but the file still isn't opened.
![Capture d’écran 1](https://github.com/pyzo/pyzo/assets/4552850/378a0523-cad4-4ebf-8b83-22f2e5f63221)

**Comment by bdieterm** [comment]
Thanks for the logger output, @operade!

Looks like recent changes (https://github.com/pyzo/pyzo/pull/923) in file "pyzo/_start.py" introduced that bug.
I will create a pull request to fix that.

Meanwhile, could you please apply the fix yourself and report back if it worked:
In file "pyzo/_start.py" replace line
```
                sys.argv[-1:] = [fname]
```
with the line
```
                sys.argv[1:] = [fname]
```
(This is actually just removing the minus sign.)

**Comment by operade** [comment]
Thank you to you @bdieterm !! It works perfectly now 
![Capture d’écran](https://github.com/pyzo/pyzo/assets/4552850/e5bdf35b-585e-4e97-9740-c6ec26597ec0)

**Comment by almarklein** [comment]
Fixed by #959
